Services/Statistics/SimulationStatistics.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
```

Comment:
we should not use this dependency here, i.e. the solution should be decoupled from the storage type (in this case Cosmos DB) so that other storage types can be used if needed

Reply:
Existing implementation (StorageRecords.cs) had cosmosdb specific implementation. I wanted to pass sql query to Get/Delete operation for all simulation stats records. If we want to avoid using SqlParameter and take a dependency on Microsoft.Azure.Documents, I can change query to get/delete one record at a time (i.e call StorageRecord.GetAsync(string id))

Reply:
StorageRecords currently supports only DocumentDb but it is designed to support other storage types. This change goes against that design principle. StorageRecords clients shouldn't need to know that DocumentDb is in use. This change will make the port of this code to RM harder, so I encourage to look for an alternative.  You can for example get the full list and filter the records client side, or implement a generic query class designed to support multiple storage engines, like StorageRecords.